semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+                            dt.now(), vp[1], vp[0], num_points_calculated
+                        )
+                    )
+                if num_points_calculated == 0:
+                    print("Shift was calculated in 0 points.", file=sys.stderr)
+                    print("Consider changing --apply_mapaxes value.")
+
+            # subsidence and displacement have opposite sign, thus the minus sign
+            ts = -self._r_factor * du * surface.dt * 1000
+            self._divide_negative_shift(ts)
+
+            ts = ts * self._convention
+
+            ts_surfaces.append(self._create_surface(ts))
+
+        return ts_surfaces
+
+    @staticmethod
+    def _vintages_name_date(vintage_pairs):
+        vintages = set()
+        for vp in vintage_pairs:
+            vintages.add(vp[0])
+            vintages.add(vp[1])
+        vintages_date = list(vintages)
+        vintages_name = []
+        for i in range(len(vintages_date)):
+            vintages_name.append("S{}".format(i))
+
+        Vintages = namedtuple("Vintages", "name date")
+        return Vintages(vintages_name, vintages_date)
+
+    def dpv(self, vintage_pairs):
+        """
+        Calulates change in pressure multiplied by cell volume
+        and sum for all cells in column
+
+        dPV must have equal sign as TS, but opposite from mathematics
+
+        monitor-base
+
+        :param vintage_pairs: list of pairs of vintages
+        :return:
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+
+        surf = self._surface
+
+        num_points = len(surf)
+        num_points_calculated = 0
+
+        shift_surfaces = []
+        pv = np.zeros((len(vintages.name), num_points))
+
+        for iv, (vn, vd) in enumerate(zip(vintages.name, vintages.date)):
+            if self._verbose:
+                print(
+                    "{:%x %X} DPV: Calculating vintage"
+                    " {:%Y.%m.%d}".format(dt.now(), vd)
+                )
+                sys.stdout.flush()
+
+            self.add_survey(vn, vd)
+            pressure = Ecl3DKW.castFromKW(
+                self._restart_views[vn]["PRESSURE"][0], self._grid
+            )
+            num_points_calculated = 0
+            for point in range(num_points):
+
+                if not np.isnan(surf.z[point]):
+                    r = surf.x[point], surf.y[point], 0
+                    try:
+                        i, j = self._grid.findCellXY(*r)
+                        num_points_calculated += 1
+                        sum_pv = 0
+                        for k in range(self._grid.getNZ()):
+                            v = self._grid.cell_volume(ijk=(i, j, k))
+                            sum_pv += pressure[i, j, k] * v
+
+                        pv[iv, point] = sum_pv
+                    except ValueError:
+                        pv[iv, point] = 0
+        for ivp, vp in enumerate(vintage_pairs):
```

Comment:
The fourth time I've seen the same logging now.. Split the functionality and have each of the other functions call it.

Reply:
Yes, 4 times as it has 4 methods for computing the surface shift. Anyway, as you suggested it'll be base logging now only.

Reply:
Made a suggestion above, and then we can

```
for ivp, vp in enumerate(vintage_pairs):
    self._report("function name", vp[0], vp[1], num_points_calculated)
``